File: H-Index - Leetcode 274/H-Index - Leetcode 274.js
var hIndex = function(citations) {
    var n = citations.length;
    var paperCounts = new Array(n + 1).fill(0);

    for (var c of citations) {
        paperCounts[Math.min(n, c)]++;
    }

    var h = n;
    var papers = paperCounts[n];

    while (papers < h) {
        h--;
        papers += paperCounts[h];
    }

    return h;
};

File: H-Index - Leetcode 274/H-Index - Leetcode 274.java
public class Solution {
    public int hIndex(int[] citations) {
        int n = citations.length;
        int[] paperCounts = new int[n + 1];

        for (int c : citations) {
            paperCounts[Math.min(n, c)]++;
        }

        int h = n;
        int papers = paperCounts[n];

        while (papers < h) {
            h--;
            papers += paperCounts[h];
        }

        return h;
    }
}
